src/shell/element/stack.rs

```diff
@@ -1169,6 +1169,27 @@ impl PointerTarget<State> for CosmicStack {
         }
     }
 
+    fn frame(&self, seat: &Seat<State>, data: &mut State) {
+        if let Some((location, serial, time)) = self
+            .0
+            .with_program(|p| p.last_location.lock().unwrap().clone())
+        {
+            self.pointer_leave_if_previous(seat, data, serial, time, location);
+        }
+
+        match self.0.with_program(|p| p.current_focus()) {
+            Focus::Header => PointerTarget::frame(&self.0, seat, data),
```

Comment:
I am not convinced this works as expected. So if the focus swaps from the application to the header/stack-elements, we send a leave to the window, but a frame to the iced-element, which will just discard it.

So the wayland client won't get a frame for this leave event. I think we need to re-work this, so that the window always gets a leave event, if it got any pointer events. Same applies to decorated windows (so `src/shell/element/window.rs`).

Reply:
`leave` on the `WlSurface` still sends a `frame` in https://github.com/Smithay/smithay/blob/509030a151ad440f8406229f32d674811efd9007/src/wayland/seat/pointer.rs#L187-L190.

This is necessary since https://github.com/Smithay/smithay/blob/509030a151ad440f8406229f32d674811efd9007/src/desktop/wayland/window.rs#L330-L334 calls `take` on the surface in `leave`, so `frame` will not be delivered to surfaces that are part of a window after that.

So focusing window decorations sends a `frame` correctly after `leave`, though this ideally could be handled in a better way.